docs/architecture/state.md

```diff
@@ -1,91 +1,111 @@
+1. Accounts
+2. Notes
+3. Nullifiers
+
+![Architecture core concepts](../img/architecture/state/state.png)
 
 ### Account database
 
-The latest account states - and data for on-chain accounts - are recorded in a sparse Merkle tree which maps account IDs to account hashes, and account data if needed.
+The accounts database stores the latest account states for public accounts or state commitments for private accounts.
 
 ![Architecture core concepts](../img/architecture/state/account-db.png)
 
 As described in the [accounts section](accounts.md), there are two types of accounts:
 
-* Public accounts where all account data is stored on-chain.
-* Private accounts where only the hashes of accounts are stored on-chain.
+- **Public accounts:** where all account data is stored on-chain.
+- **Private accounts:** where only the hash of the account is stored on-chain.
 
-Private accounts significantly reduce the storage overhead for nodes. A private account contributes only $40$ bytes to the global state ($8$ bytes account ID + $32$ bytes account hash). Or, said another way, 1 billion private accounts takes up only $40$ GB of state.
+Private accounts significantly reduce storage overhead. A private account contributes only $40$ bytes to the global state ($8$ bytes for the account ID + $32$ bytes for the account hash). For example, 1 billion private accounts take up only $40$ GB of state.
+
+The storage contribution of a public account depends on the amount of data it stores.
 
 > **Warning**
->  Losing the state of a private account means loss of funds in a similar manner as a loss of a private key - as the user won't be able to execute transactions. This problem can be mitigated by storing encrypted account state in the cloud or backing it up somewhere else. Unlike storing private keys in the cloud, this does not compromise privacy or account security.
->
->  In the future we hope to enable encrypted accounts where the account data is stored on-chain but in an encrypted format. This is especially interesting for shared accounts like advanced multi-sig wallets.
+> - In Miden, when the user is the custodian of their account state (in the case of a private account), losing this state amounts to losing their funds, similar to losing a private key.
 
 ### Note database
 
-Notes are recorded in an append-only accumulator, a [Merkle Mountain Range](https://github.com/opentimestamps/opentimestamps-server/blob/master/doc/merkle-mountain-range.md). Each leaf is a block header which contains the commitment to all notes created in that block. The commitment is a Sparse Merkle Tree of all the notes in a block. The size of the Merkle Mountain Range grows logarithmically with the number of items in it.
```

Comment:
I would mention the append-only accumulator here - it is a pretty important architectural detail (though the description here is not correct - see previous comment).